crates/network/src/any/mod.rs

```diff
@@ -46,11 +68,537 @@ impl From<TxType> for AnyTxType {
+                if fields.contains_key("authorizationList") {
+                    Some(memo.authorization_list.get_or_init(|| {
+                        self.deser_by_key("authorizationList")
+                            .and_then(Result::ok)
+                            .unwrap_or_default()
+                    }))
+                } else {
+                    None
+                }
+            }
+        }
+    }
+}
+
+/// Unsigned transaction type for a catch-all network.
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+#[doc(alias = "AnyTypedTx")]
+pub enum AnyTypedTransaction {
+    /// An Ethereum transaction.
+    Ethereum(TypedTransaction),
+    /// A transaction with unknown type.
+    Other {
+        #[serde(rename = "type")]
+        /// Transaction type.
+        ty: AnyTxType,
+        /// Additional fields.
+        fields: std::collections::BTreeMap<String, serde_json::Value>,
+
+        /// Memoization for deserialization.
+        #[serde(skip, default)]
+        memo: DeserMemo,
+    },
+}
+
+impl AnyTypedTransaction {
+    /// Select a field by key and attempt to deserialize it.
+    ///
+    /// This method will return `None` if the key is not present in the fields,
+    /// or if the transaction is already fully deserialized (i.e. it is an
+    /// Ethereum [`TxEnvelope`]). Otherwise, it will attempt to deserialize the
+    /// field and return the result wrapped in a `Some`.
+    pub fn deser_by_key<T: serde::de::DeserializeOwned>(
+        &self,
+        key: &str,
+    ) -> Option<serde_json::Result<T>> {
+        let Self::Other { fields, .. } = self else {
+            return None;
+        };
+
+        fields.get(key).cloned().map(serde_json::from_value)
+    }
+}
+
+impl From<TypedTransaction> for AnyTypedTransaction {
+    fn from(value: TypedTransaction) -> Self {
+        Self::Ethereum(value)
+    }
+}
+
+impl From<AnyTxEnvelope> for AnyTypedTransaction {
+    fn from(value: AnyTxEnvelope) -> Self {
+        match value {
+            AnyTxEnvelope::Ethereum(tx) => Self::Ethereum(tx.into()),
+            AnyTxEnvelope::Other { ty, fields, memo, .. } => Self::Other { ty, fields, memo },
+        }
+    }
+}
+
+impl From<AnyTypedTransaction> for WithOtherFields<TransactionRequest> {
+    fn from(value: AnyTypedTransaction) -> Self {
+        match value {
+            AnyTypedTransaction::Ethereum(tx) => Self::new(tx.into()),
+            AnyTypedTransaction::Other { ty, mut fields, .. } => {
+                fields.insert("type".to_string(), serde_json::Value::Number(ty.0.into()));
+                Self { inner: Default::default(), other: OtherFields::new(fields) }
+            }
+        }
+    }
+}
+
+impl From<AnyTxEnvelope> for WithOtherFields<TransactionRequest> {
+    fn from(value: AnyTxEnvelope) -> Self {
+        AnyTypedTransaction::from(value).into()
+    }
+}
+
+impl alloy_consensus::Transaction for AnyTypedTransaction {
```

Comment:
this duplicates a lot of things that we also have on the anyevelope.
we could simplify this if we convert type + fields + memo into it's own type.

the it also becomes possible to impl TryFrom conversions, e.g. for OpDeposit tx

Reply:
done